File: Customer Account Tracker Backend/customer_account_tracker/src/main/java/com/cts/customer_account_tracker/Service/UserService.java
package com.cts.customer_account_tracker.Service;

import com.cts.customer_account_tracker.Entity.User;

public interface UserService {
	
	public User registerUser(User user);
	public User getUserById(String userId);
	public String deleteUserById(String userId);
	public User updateUser(String userId,User user);
	public User authenticateUser(String userId, String password);
}

File: FrontEnd/customeraccounttrackerapp/src/services/UserService.js
import axios from 'axios';

const USER_BASE_REST_API_URL = "http://localhost:8081/users";
class UserService{

    registerUser(user) {
        return axios.post(USER_BASE_REST_API_URL, user);
    }

    getUserById(userId) {
        return axios.get(USER_BASE_REST_API_URL+"/"+userId);
    }

    updatePassword(userId,user) {
        return axios.put(`${USER_BASE_REST_API_URL}/${userId}`,user)
    }

    deleteUser(userId) {
        return axios.delete(`${USER_BASE_REST_API_URL}/${userId}`)
    }
    authenticateUser(userId, password) {
        return axios.post(`${USER_BASE_REST_API_URL}/login?userId=${userId}&password=${password}`)
    }
}

export default new UserService();